src/java/org/apache/cassandra/utils/ByteBufferUtil.java

```diff
@@ -927,6 +949,19 @@ private static boolean startsWith(ByteBuffer src, ByteBuffer prefix, int offset)
         return true;
     }
 
+    public static <T> ByteBuffer serialized(IVersionedSerializer<T> serializer, T value, int version)
+    {
+        try (DataOutputBuffer dob = new DataOutputBuffer())
+        {
+            serializer.serialize(value, dob, version);
+            return dob.buffer();
+        }
+        catch (IOException e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
+
     public static final IVersionedSerializer<ByteBuffer> vintSerializer = new IVersionedSerializer<ByteBuffer>()
```

Comment:
naming: Maybe something like `bytesSerializer`, given the vint part is an implementation detail (i.e. how we write the length)?

Reply:
Renamed to `byteBufferSerializer` and `nullableByteBufferSerializer`